contents/blog/dataquality_expectations_facet/index.mdx

```diff
@@ -0,0 +1,410 @@
+justify bad decisions by claiming that they are “data-based” without ever validating the quality of our sources. Where
+did that data come from? How old is it? Is the logic that generated it correct? Is it derived from some other dataset?
+What is the quality of that dataset?
+![Data-driven: You keep using that word. I do not think it means what you think it means](./data_driven.png)
+
+Thankfully, data quality validation is becoming more and more common in data engineering organizations. In part, this is
+due to the prevalence of new tools and their integration with common workflow engines which we already use to schedule
+the jobs that generate and process our data. One such tool that has been gaining in popularity is called [_Great
+Expectations_](https://docs.greatexpectations.io), a Python-based framework for defining assertions about data sets
+which easily integrates with existing workflow tools, such as Airflow. 
+
+In software development, testing the behavior of our code with unit and integration tests has been common practice for
+years. Similarly, using Great Expectations, a data engineer can assert that a dataset has a row count that falls within
+an expected range, that column values are not null, or that values match a specified regular expression. One can even
+create custom expectations, such as validating that the number of records in treatment is roughly the same as the number
+of records in control (this post is not intended to be an in-depth tutorial on setting up Great Expectations; if you
+want to read more on its capabilities and to get started, I recommend the going through
+the [Quick Start tutorial](https://docs.greatexpectations.io/en/latest/guides/tutorials/quick_start.html)).
+
+### A Sample Assertion Suite
+
+As a simple example, imagine a table of new customers that you need to import into your Data Warehouse. Before
+importing, we want to check the data quality of this new batch of customers. One example suite of assertions we could
+test is below:
+
+```json
+{
+  "data_asset_type": "Dataset",
+  "expectation_suite_name": "customers_suite",
+  "expectations": [
+    {
+      "expectation_type": "expect_table_row_count_to_be_between",
+      "kwargs": {
+        "max_value": 1000,
+        "min_value": 100
+      },
+      "meta": {}
+    },
+    {
+      "expectation_type": "expect_table_column_count_to_equal",
+      "kwargs": {
+        "value": 8
+      },
+      "meta": {}
+    },
+    {
+      "expectation_type": "expect_table_columns_to_match_ordered_list",
+      "kwargs": {
+        "column_list": [
+          "id",
+          "created_at",
+          "updated_at",
+          "name",
+          "email",
+          "address",
+          "phone",
+          "city_id"
+        ]
+      },
+      "meta": {}
+    },
+    {
+      "expectation_type": "expect_column_values_to_be_unique",
+      "kwargs": {
+        "column": "email"
+      },
+      "meta": {}
+    }
+  ],
+  "meta": {
+    // ...
```

Comment:
Does it also show up in red once rendered by gatsby? maybe not.

Reply:
Looks like this in Gatsby
<img width="935" alt="Screen Shot 2021-08-10 at 10 26 48 AM" src="https://user-images.githubusercontent.com/40346148/128906580-18e48d10-aaa7-4d89-b868-546ea7e6a9ed.png">

Reply:
looks good